src/main/kotlin/io/github/inductiveautomation/kindling/idb/generic/QueryResult.kt

```diff
@@ -18,7 +19,13 @@ sealed interface QueryResult {
         override fun getColumnCount(): Int = columnNames.size
         override fun getColumnName(columnIndex: Int): String = columnNames[columnIndex]
         override fun getColumnClass(columnIndex: Int): Class<*> = columnTypes[columnIndex]
-        override fun getValueAt(rowIndex: Int, columnIndex: Int): Any? = data[rowIndex][columnIndex]
+        override fun getValueAt(rowIndex: Int, columnIndex: Int): Any? {
+            val row = data[rowIndex]
+            return when (columnIndex) {
+                0 -> Date(row[0] as Long) // "timestmp" column, convert Long to Date
```

Comment:
I liked your first fix in https://github.com/paul-griffith/kindling/pull/174 better 😄 
This will only work if the first column happens to be a timestamp-as-long, right? (plus it'll blow up if the first column is any other data type). I think there's definitely something to do better here, but it might have to involve adjusting the data during `QueryResult` construction, where the `Timestamp` type columns are coming from.